### ISSUE TITLE
Nade Trajectory

### ISSUE THREAD

**Issue by thiagofelix** [StatsHelix/demoinfo#139]
Is it possible to grab the grenade trajectory from point where it is fired to the point where it explodes?

**Comment by econoraptor** [comment]
Yes, it's possible.  Each grenade has an associated projectile entity that you can track in the air.  HEs and flashes are both `CBaseCSGrenadeProjectile`, molotovs and incendiaries use  `CMolotovProjectile` and the others use `CSmokeGrenadeProjectile` `CDecoyProjectile`.  You can track their positions using the `Cell` properties.  If you want to see an implementation, look here: https://github.com/econoraptor/demoinfo/tree/DetonateInterpolate2

**Comment by moritzuehling** [comment]
We'd be open to a PR that opens the grenade-projectiles up via the API.

**Comment by thiagofelix** [comment]
@econoraptor thanks for the explanation, I've been trying to project their trajectory using grenades velocity, angle, position at the throw moment but I think the cell strategy is more bullet proof.

**Comment by moritzuehling** [comment]
@thiagofelix That method might have worked for demos where the snapshotrate matched the tickrate, but it would have been fairly inaccurate for all other demos.

**Comment by thiagofelix** [comment]
@moritzuehling you'r right, just tried out using the cells and it worked like a charm =) thank you so much.

I am building a replay viewer that parses the demo in the browser itself on user's computer, sort of standalone app without need of a server, so far got everything I wanted working (player's position, nades, memory consumption acceptable)

Thanks again for the help
